app/build.gradle

```diff
@@ -124,8 +124,10 @@ dependencies {
     implementation 'com.google.guava:guava:28.1-jre'
     annotationProcessor 'com.google.guava:guava:24.1-jre'
     implementation 'org.apache.commons:commons-lang3:3.9'
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
+    implementation 'com.vipulasri:ticketview:1.0.7'
```

Comment:
this lib looks at least not unmaintained... Usually I am not so happy with adding new dependencies, but why not. Doing it from scratch is also not worth the effort. But please make sure to mention it in the list of used libraries

Reply:
Constrain Layout is the most common layout now. The another one, "ticketView" is because of this semicircle.